pkgs/development/interpreters/perl/default.nix

```diff
@@ -101,8 +101,21 @@ let
     enableParallelBuilding = !crossCompiling;
 
     preConfigure = ''
-        substituteInPlace ./Configure --replace '`LC_ALL=C; LANGUAGE=C; export LC_ALL; export LANGUAGE; $date 2>&1`' 'Thu Jan  1 00:00:01 UTC 1970'
-        substituteInPlace ./Configure --replace '$uname -a' '$uname --kernel-name --machine --operating-system'
+        # perl includes the build date, the uname of the build system and the
+        # username of the build user in some files.
+        # We override these to make it build deterministically.
+        # Very much inspired by what Debian did for reproducibility: https://salsa.debian.org/perl-team/interpreter/perl/blob/debian-5.26/debian/config.over
+        # other distro solutions
+        # https://github.com/bmwiedemann/openSUSE/blob/master/packages/p/perl/perl-reproducible.patch
+        # https://github.com/archlinux/svntogit-packages/blob/packages/perl/trunk/config.over
+        # A ticket has been opened upstream to possibly clean some of this up: https://rt.perl.org/Public/Bug/Display.html?id=133452
```

Comment:
I think if we move this before the preConfigure we can change it in the future without triggering a rebuild. Maybe this is a good thing for such a long comment? I am not sure.